closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -932,7 +931,9 @@ private void initConstraints() {
          * @return true if the Service is allowed
          */
         boolean isRestrictedServiceAllowed(Service service) {
-            String providerName = service.getProvider().getName();
+            Provider provider = service.getProvider();
+            String providerName = provider.getName();
+            String providerClassName = provider.getClass().getName();
```

Comment:
Suggest updating the local in `isProviderAllowed(Class<?>)` from `providerName` to `providerClassName` to be consistent with this method.